### ISSUE TITLE
Is it possible to track the attention (embeddings) through fine tunning process

### ISSUE THREAD

**Issue by Coolchong** [vandijklab/cell2sentence#8]
Hello,

Very cool model! I tried it with our data set, and it gave great accuracy in terms of cell type prediction. I am wondering, since I am not trained as a data scientist, how could I extract the attentions of the sentences or embeddings (I hope these are the correct terms) during the fine tuning process to understand how C2S make predictions.

Thanks!

**Comment by SyedA5688** [comment]
Thanks, great to hear that it worked well on your data!
For extracting attention coefficients from the model, the easiest way is to use the `output_attentions` option in model.generate(). The documentation for that forward pass argument is located here:
https://huggingface.co/docs/transformers/en/main_classes/text_generation#transformers.GenerationConfig.output_attentions

To use it, you would need to pass `output_attentions=True` in the call to the model. Many of our functions pass kwargs to model.generate() under the hood, so you should be able to do something like:
```
csmodel = ...  # Load data
csdata = ...  # Load model

predicted_cell_types = predict_cell_types_of_data(
    csdata=csdata,
    csmodel=csmodel,
    n_genes=200,
    output_attentions=True
)
```
and the code will pass the argument to model.generate(). A bit of code modification might be needed, since the output object returned by the model will now return attention coefficient matrices as well:
```
outputs = model.generate(
    input_ids=input_ids,
    attention_mask=attention_mask,
    max_new_tokens=max_num_tokens,
    pad_token_id=self.tokenizer.pad_token_id,
    **kwargs
)
```
Here, the `output` object will contain logits from the LLM as normal, as well as attention coefficients, so you will nee to be careful about how you access attributes of the objects in subsequent code. Hope this helps!
